datasafe-business-impl/src/test/java/de/adorsys/datasafe/business/impl/cmsencryption/CmsEncryptionServiceImplTest.java

```diff
@@ -31,38 +39,114 @@
 
+    private static final String TEST_MESSAGE_CONTENT = "message content";
+
+    private static KeyStoreAccess keyStoreAccess;
+    private static KeyStoreService keyStoreService = new KeyStoreServiceImpl();
+
     private CMSEncryptionService cmsEncryptionService = new CMSEncryptionServiceImpl();
-    private KeyStoreService keyStoreService = new KeyStoreServiceImpl();
-    private static final String MESSAGE_CONTENT = "message content";
+
+    @BeforeAll
+    public static void setUp() {
+        keyStoreAccess = getKeyStoreAccess();
+    }
 
     @Test
     @SneakyThrows
     public void cmsStreamEnvelopeEncryptAndDecryptTest() {
-        ReadKeyPassword readKeyPassword = new ReadKeyPassword("readkeypassword");
-        ReadStorePassword readStorePassword = new ReadStorePassword("readstorepassword");
-        KeyStoreAuth keyStoreAuth = new KeyStoreAuth(readStorePassword, readKeyPassword);
-
-        KeyStoreCreationConfig config = new KeyStoreCreationConfig(1, 0, 1);
-        KeyStore keyStore = keyStoreService.createKeyStore(keyStoreAuth, KeyStoreType.DEFAULT, config);
-
-        KeyStoreAccess keyStoreAccess = new KeyStoreAccess(keyStore, keyStoreAuth);
+        KeyStoreAccess keyStoreAccess = getKeyStoreAccess();
 
         PublicKeyIDWithPublicKey publicKeyIDWithPublicKey = keyStoreService.getPublicKeys(keyStoreAccess).get(0);
         ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
 
         OutputStream encryptionStream = cmsEncryptionService.buildEncryptionOutputStream(outputStream,
                 publicKeyIDWithPublicKey.getPublicKey(), publicKeyIDWithPublicKey.getKeyID());
 
-        encryptionStream.write(MESSAGE_CONTENT.getBytes());
+        encryptionStream.write(TEST_MESSAGE_CONTENT.getBytes());
         encryptionStream.close();
         byte[] byteArray = outputStream.toByteArray();
 
         ByteArrayInputStream inputStream = new ByteArrayInputStream(byteArray);
         InputStream decryptionStream = cmsEncryptionService.buildDecryptionInputStream(inputStream, keyStoreAccess);
         byte[] actualResult = IOUtils.toByteArray(decryptionStream);
 
-        assertThat(MESSAGE_CONTENT).isEqualTo(new String(actualResult));
-        log.debug("en and decrypted successfully");
+        assertThat(TEST_MESSAGE_CONTENT).isEqualTo(new String(actualResult));
+    }
+
+    @Test
+    @SneakyThrows
+    public void cmsStreamEnvelopeZeroKeyPairFailTest() {
+        PublicKeyIDWithPublicKey publicKeyIDWithPublicKey = keyStoreService.getPublicKeys(keyStoreAccess).get(0);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        CMSEnvelopedDataStreamGenerator gen = new CMSEnvelopedDataStreamGenerator();
+        gen.open(outputStream, new JceCMSContentEncryptorBuilder(CMSAlgorithm.AES128_CBC)
+                .setProvider(BouncyCastleProvider.PROVIDER_NAME).build());
+
+        OutputStream encryptionStream = cmsEncryptionService.buildEncryptionOutputStream(outputStream,
```

Comment:
maybe try with resources may shorten this a bit

Reply:
ok, will fix